### ISSUE TITLE
vscode-file: running out of sources fails to load sourcemaps

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#120663]
Steps: just run out of sources with `vscode-file` enabled.

It looks like devtools fails to load sourcemaps with `vscode-file` URL but I cannot really understand how that is possible unless the protocol handler we install is not being used from devtools:

```
DevTools failed to load SourceMap: Could not parse content for vscode-file://vscode-app/Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/contrib/notebook/browser/view/renderers/markdownCell.js.map: Unexpected end of JSON input
```

![image](https://user-images.githubusercontent.com/900690/113764903-e2ce1580-971b-11eb-83e4-c32f8133516c.png)

**Comment by bpasero** [comment]
Proof that the URLs work from devtools, running:

```
await (await fetch ("vscode-file://vscode-app/Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/contrib/notebook/browser/view/renderers/markdownCell.js.map")).json()
```

Properly returns a JSON object:

![image](https://user-images.githubusercontent.com/900690/113816826-a8916200-9775-11eb-8c35-42a970f80500.png)

So I am really puzzled why it would not work.

**Comment by bpasero** [comment]
Looks like https://github.com/electron/electron/issues/21699 is the culprit here. 

Similar issue we had closed before: https://github.com/microsoft/vscode/issues/115440

**Comment by bpasero** [comment]
One mitigation idea from @connor4312 is to enable inline sourcemaps.

**Comment by deepak1556** [comment]
The network call from devtools frontend to load source map will eventually delegate to the runtime, it is currently handled in Electron at https://github.com/electron/electron/blob/12-x-y/shell/browser/ui/inspectable_web_contents.cc#L646. The problem is that the URLLoaderFactory created will not understand custom protocols hence the response stream is incorrect.

**Comment by deepak1556** [comment]
Fixed in 3fb483e7fff42559b5f93e8aa92d0354021ee5ff
